File: example/src/main/java/com/github/simonpercic/oklogexample/ui/MainActivity.java
package com.github.simonpercic.oklogexample.ui;

import android.os.Bundle;
import android.support.design.widget.CoordinatorLayout;
import android.support.design.widget.Snackbar;
import android.view.View;

import com.github.simonpercic.oklogexample.R;
import com.github.simonpercic.oklogexample.ui.base.DaggerActivity;

import java.util.List;

import javax.inject.Inject;

import butterknife.BindView;
import butterknife.BindViews;
import butterknife.ButterKnife;
import butterknife.OnClick;

/**
 * @author Simon Percic <a href="https://github.com/simonpercic">https://github.com/simonpercic</a>
 */
public class MainActivity extends DaggerActivity<MainComponent> implements MainMvp.View {

    private static final ButterKnife.Setter<View, Boolean> ENABLE_VIEW = (view, value, index) -> view.setEnabled(value);

    @BindView(R.id.activity_root_coordinator_layout) CoordinatorLayout coordinatorLayout;
    @BindViews({R.id.btn_get, R.id.btn_post, R.id.btn_put, R.id.btn_delete, R.id.btn_header}) List<View> buttonViews;

    @Inject MainMvp.Presenter presenter;

    // region Activity lifecycle callbacks

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        ButterKnife.bind(this);
    }

    @Override protected void onDestroy() {
        super.onDestroy();
        presenter.onDestroy();
    }

    // endregion Activity lifecycle callbacks

    // region Abstract impl

    @Override protected MainComponent inject() {
        MainComponent component = DaggerMainComponent.builder()
                .appComponent(getAppComponent())
                .mainModule(new MainModule(this))
                .build();

        component.inject(this);

        return component;
    }

    // endregion Abstract impl

    // region Click listeners

    @OnClick(R.id.btn_get)
    public void onBtnGetClicked() {
        presenter.onBtnGetClicked();
    }

    @OnClick(R.id.btn_post)
    public void onBtnPostClicked() {
        presenter.onBtnPostClicked();
    }

    @OnClick(R.id.btn_put)
    public void onBtnPutClicked() {
        presenter.onBtnPutClicked();
    }

    @OnClick(R.id.btn_delete)
    public void onBtnDeleteClicked() {
        presenter.onBtnDeleteClicked();
    }

    @OnClick(R.id.btn_header)
    public void onBtnHeaderClicked() {
        presenter.onBtnHeaderClicked();
    }

    // endregion Click listeners

    // region Helper methods

    @Override
    public void disableButtons() {
        ButterKnife.apply(buttonViews, ENABLE_VIEW, false);
    }

    @Override
    public void enableButtons() {
        ButterKnife.apply(buttonViews, ENABLE_VIEW, true);
    }

    @Override
    public void showMessage(String message) {
        Snackbar.make(coordinatorLayout, message, Snackbar.LENGTH_SHORT).show();
    }

    // endregion Helper methods
}

File: example/src/main/kotlin/com/github/simonpercic/oklogexample/ui/MainActivity.kt
package com.github.simonpercic.oklogexample.ui

import android.os.Bundle
import android.support.design.widget.Snackbar
import com.github.simonpercic.oklogexample.R
import com.github.simonpercic.oklogexample.ui.base.DaggerActivity
import kotlinx.android.synthetic.main.activity_main.*
import javax.inject.Inject

/**
 * @author Simon Percic [https://github.com/simonpercic](https://github.com/simonpercic)
 */
class MainActivity : DaggerActivity<MainComponent>(), MainMvp.View {

    @Inject internal lateinit var presenter: MainMvp.Presenter

    // region Activity lifecycle callbacks

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        btnGet.setOnClickListener { presenter.onBtnGetClicked() }
        btnPost.setOnClickListener { presenter.onBtnPostClicked() }
        btnPut.setOnClickListener { presenter.onBtnPutClicked() }
        btnDelete.setOnClickListener { presenter.onBtnDeleteClicked() }
        btnHeader.setOnClickListener { presenter.onBtnHeaderClicked() }
    }

    override fun onDestroy() {
        super.onDestroy()
        presenter.onDestroy()
    }

    // endregion Activity lifecycle callbacks

    // region Abstract impl

    override fun inject(): MainComponent {
        val component = DaggerMainComponent.builder()
            .appComponent(appComponent)
            .mainModule(MainModule(this))
            .build()

        component.inject(this)

        return component
    }

    // endregion Abstract impl

    // region View impl

    override fun disableButtons() {
        setButtonsEnabled(false)
    }

    override fun enableButtons() {
        setButtonsEnabled(true)
    }

    override fun showMessage(message: String) {
        Snackbar.make(rootCoordinatorLayout, message, Snackbar.LENGTH_SHORT).show()
    }

    // endregion View impl

    // region Helper methods

    private fun setButtonsEnabled(enabled: Boolean) {
        btnGet.isEnabled = enabled
        btnPost.isEnabled = enabled
        btnPut.isEnabled = enabled
        btnDelete.isEnabled = enabled
        btnHeader.isEnabled = enabled
    }

    // endregion Helper methods
}
